### ISSUE TITLE
Add "ES_APPLICATION_LOGS" log_type for aws_elasticsearch_domain

### ISSUE THREAD

**Issue by smelchior** [terraform-providers/terraform-provider-aws#5411]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS now allows to export the application logs of ES to Cloudwatch. I want to configure this with terraform.

### New or Affected Resource(s)

* aws_elasticsearch_domain

### Potential Terraform Configuration

```hcl
resource "aws_elasticsearch_domain" "es" {
.....
log_publishing_options {
log_type = "ES_APPLICATION_LOGS"
cloudwatch_log_group_arn  = "/...."
}
....
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2018/07/troubleshoot_your_amazon_elasticsearch_service_domains_easily_using_error_logs/

**Comment by bflad** [comment]
Support for this has been merged in and will release with version 1.31.0 of the AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.31.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1310-august-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
